### PR TITLE
Tidy up names, expand tlas and abbvns, correct docstrings, and refactor the magnification dialog's constructor

### PR DESCRIPTION
**Pull request by samcunliffe**
Some tidying and light refactoring.

### CODE REVIEW

**Comment by Joseph-Garvey** [comment]
It would probably be sensible to combine PR #129 here as well.

**Comment by Joseph-Garvey** [comment]
Added changes from PR #129 , however this branch is using the convention of btn_(item) instead of what has been used here (item_btn).
Would be good to get the two to be the same, if you have strong feelings either way let me know and I can go ahead and make the change.

**Comment by Joseph-Garvey** [comment]
@samcunliffe Will try get this PR merged first, as the major refactoring changes have been creating merge conflicts with main for the last week a huge pain - (i have learned a lesson here, that i really ought to be pushing refactoring changes first BEFORE doing anything else!)

**Comment by Joseph-Garvey** [comment]
@samcunliffe Happy to merge this?

**Comment by samcunliffe** [comment]
> It would probably be sensible to combine PR #129 here as well.

I'm a bit late to the party no (and definitely don't undo it) but I'd probably have kept variable renaming in a separate PR than refactoring the constructors.  Because: separate scope.

**Comment by samcunliffe** [comment]
> Added changes from PR #129 , however this branch is using the convention of btn_(item) instead of what has been used here (item_btn). Would be good to get the two to be the same, if you have strong feelings either way let me know and I can go ahead and make the change.

 `btn_something` or `something_button`? For me the latter is much clearer because it saves the miliseconds of mental overhead to add the vowels back in to `btn`. And we're not line-length limited very much.

File: src/cavendish_particle_tracks/_magnification_dialog.py [hunk 1]
```diff
@@ -16,12 +20,16 @@
...
 
 class MagnificationDialog(QDialog):
     def __init__(self, parent=None):
-        super().__init__(parent)
-
-        self.parent = parent
+        super().__init__(parent)  # Call QDialog constructor
+        self.parent: ParticleTracksWidget = (
+            parent  # Assign Napari viewer as parent.
+        )
```
Comment (samcunliffe) [inline src/cavendish_particle_tracks/_magnification_dialog.py:32]: Can we get around this by putting the type hints in the argument?

```python
def __init__(self, parent: QWidget | None = None):
    super().__init__(parent)
    self.parent = parent
```

Actually, now I come to think of it, does the super classes constructor not set the parent for us? TBC

File: src/cavendish_particle_tracks/_magnification_dialog.py [hunk 2]
```diff
@@ -30,77 +38,99 @@ def __init__(self, parent=None):
...
         self.b2 = Fiducial()
 
-        # drop-down lists of fiducials
-        self.cbf1 = self._setup_dropdown_fiducials_combobox()
-        self.cbf2 = self._setup_dropdown_fiducials_combobox()
-        self.cbb1 = self._setup_dropdown_fiducials_combobox(back=True)
-        self.cbb2 = self._setup_dropdown_fiducials_combobox(back=True)
+        # region UI Setup
+        self.ui_setup()
+        self.cal_layer = self.create_or_retrieve_magnification_layer()
```
Comment (samcunliffe) [inline src/cavendish_particle_tracks/_magnification_dialog.py:43]: Might be worth renaming that to `self.magnification_layer` whilst we're in a renaming mood.

File: src/cavendish_particle_tracks/_magnification_dialog.py [hunk 3]
```diff
@@ -30,77 +38,99 @@ def __init__(self, parent=None):
...
-
+        # Add selected fiducial buttons
+        self.btn_add_f1 = QPushButton("Add")
+        self.btn_add_f2 = QPushButton("Add")
+        self.btn_add_b1 = QPushButton("Add")
+        self.btn_add_b2 = QPushButton("Add")
+        self.btn_add_f1.clicked.connect(self._on_click_add_coords_f1)
+        self.btn_add_f2.clicked.connect(self._on_click_add_coords_f2)
+        self.btn_add_b1.clicked.connect(self._on_click_add_coords_b1)
+        self.btn_add_b2.clicked.connect(self._on_click_add_coords_b2)
```
Comment (samcunliffe) [inline src/cavendish_particle_tracks/_magnification_dialog.py:78]: Hmm. `add_f1_marker_button`? `add_front1_fiducial_button`? Which is clearest? @palvarezc any opinion?

File: src/cavendish_particle_tracks/_magnification_dialog.py [hunk 4]
```diff
@@ -30,77 +38,99 @@ def __init__(self, parent=None):
...
+        # Calculate magnification button
+        self.btn_calculate = QPushButton("Calculate magnification")
+        self.btn_calculate.clicked.connect(self._on_click_magnification)
```
Comment (samcunliffe) [inline src/cavendish_particle_tracks/_magnification_dialog.py:81]: For example if you name the variable well, you can do away with the comment...
```suggestion
        self.calculate_magnification_button = QPushButton("Calculate magnification")
        self.calculate_magnification_button.clicked.connect(self._on_click_magnification)
```

Which makes [CodeAesthetic happy](https://www.youtube.com/watch?v=Bf7vDBBOBUA).

File: src/cavendish_particle_tracks/_magnification_dialog.py [hunk 5]
```diff
@@ -113,22 +143,22 @@ def _setup_dropdown_fiducials_combobox(self, back=False):
...
     def _on_click_add_coords_f1(self) -> None:
         """Add first front fiducial"""
-        self.f1.name = self.cbf1.currentText()
+        self.f1.name = self.cmb_front1.currentText()
         self.f1.x, self.f1.y = self._add_coords(0)
 
     def _on_click_add_coords_f2(self) -> None:
         """Add second front fiducial"""
-        self.f2.name = self.cbf2.currentText()
+        self.f2.name = self.cmb_front2.currentText()
```
Comment (samcunliffe) [inline src/cavendish_particle_tracks/_magnification_dialog.py:151]: I've already forgotten what `cb` and `cmb` are 🙃 . Combo box?

**Review by samcunliffe — COMMENTED** [review]
Yeah nearly ready. I have some more thoughts and could add a bit over the weekend? Though I suspect it's already good enough to push ahead and merge approximately as-is, it's an improvement in any case. So we can open for review now if you want to get things closed up...

I leave to you, @Joseph-Garvey

**Comment by Joseph-Garvey** [comment]
> > Added changes from PR #129 , however this branch is using the convention of btn_(item) instead of what has been used here (item_btn). Would be good to get the two to be the same, if you have strong feelings either way let me know and I can go ahead and make the change.
> 
> `btn_something` or `something_button`? For me the latter is much clearer because it saves the miliseconds of mental overhead to add the vowels back in to `btn`. And we're not line-length limited very much.

I think the reason I've used this historically is for intellisense completion - I just need to type btn_r... and it'll complete. Whereas radius_( ) will have multiple UI elements associated with it and so take longer to complete.
But for the sake of clarity I'll make this change.

File: src/cavendish_particle_tracks/_magnification_dialog.py (hunk 1 above)
Comment (Joseph-Garvey) [reply src/cavendish_particle_tracks/_magnification_dialog.py:32]: Using :QWidget as a type hint would stop the type hinter from recognising particle-tracks-widget-specific variables and methods, which I've been trying to keep for easy debugging and ensuring merges don't go sideways.
Using :ParticleTracksWidget here stops the tests from running, as it appears that when the test linter goes through it doesn't pick up the TYPE_CHECKING statement and throws a "ParticleTracksWidget not recognised" error.

File: src/cavendish_particle_tracks/_magnification_dialog.py (hunk 5 above)
Comment (Joseph-Garvey) [reply src/cavendish_particle_tracks/_magnification_dialog.py]: changed `cmb` to `combobox`

File: src/cavendish_particle_tracks/_magnification_dialog.py (hunk 3 above)
Comment (palvarezc) [reply src/cavendish_particle_tracks/_magnification_dialog.py]: Maybe `add_front1_fiducial_button`

**Comment by palvarezc** [comment]
Is this ready for review?

**Review by palvarezc — APPROVED** [review]

**Comment by Joseph-Garvey** [comment]
Whilst there is more that can be done here, I'm keen to get functionality complete before I leave. 
The more the main branch shifts whilst I try and get it done, the more work it's creating for me in resolving merge conflicts and dealing with different conventions between branches.

File: src/cavendish_particle_tracks/_magnification_dialog.py (hunk 4 above)
Comment (Joseph-Garvey) [reply src/cavendish_particle_tracks/_magnification_dialog.py]: https://www.youtube.com/watch?v=Bf7vDBBOBUA was a very good recommendation. Will keep it in mind going forward!

**Pull request merged**